Add gtnh egg

DJSchaffner opened the pull request:
# Description

<!-- Please explain what is being changed or added as a short overview for this PR. Also, link existing relevant issues if they exist with resolves # -->

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done (i.e. [x]). You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/pelican-eggs/minecraft/blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
    * There is a kind of stale PR open with the same subject. More info down below
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

## Information

I originally added this as PR for [this](https://github.com/Ptero-Eggs/game-eggs/pull/112) other community repo since I couldn't find an egg that supports GT:NH, however it has just been brought to my attention that I should probably use this repo, even though I am personally not using Pelican (yet) since this is the more official one.

Thus I now ported the PR with some minor changes.

Also regarding the second open PR #59  for GT:NH: This egg features downgrade/upgrade support for other versions, cleaning up game files but retaining the world and lets you install beta version as well as fully released versions from the devs website (specific ones or just 'latest').
Also, there are some logs in the install script that should help identify issue if there are any when setting faulty game version or else, as well as some other minor differences in startup etc.

Feel free to merge this or the other one as you see fit.

DJSchaffner committed:
Add gtnh egg json and readmes

samsonsin commented:
[Following steps here during update/install](https://wiki.gtnewhorizons.com/wiki/Server_Setup) would be prudent. Changing removal from a blacklist of folders, instead whitelist;

config (exluding JourneyMapServer), libraries, mods, resources (if present), scripts (if present), lwjgl3ify-forgePatches.jar, java9args.txt.

DJSchaffner committed:
Add restore mechanism for JourneyMapServer

DJSchaffner commented:
> [Following steps here during update/install](https://wiki.gtnewhorizons.com/wiki/Server_Setup) would be prudent.
Added restore mechanism for JourneyMapServer. Not sure if backing up the whole container is such a necessity since most folders and files are not really meant to be edited and additional mods will have to be added manually anyway

Can look into swapping out the blacklist for a whitelist later

samsonsin commented:
Backing up entire server is just good praxis when doing it all manually, in case you mess up. In this case, so long as upgrades/downgrades work correctly it's not an issue. I would probably just make a full backup using the panel, this is nothing you would worry about in the install script I believe.

i see you made the script restore the journey map config, but elected to remain with the blacklist. Probably largely interchangeable, though personally I feel only deleting what you must is probably better than deleting everything other than what you need. Probably largely a matter of taste rather than an actual benefit. On one hand, on method is probably safer/recommended. The other is already implemented and while perfection is fine, rewrites are still work that need testing and all that. You decide

DJSchaffner committed:
Remove config backup after install

DJSchaffner commented:
We discussed a bit and tried some stuff and I decided that the way it is right now it is good enough. Sure there can be more improvements and customizability however, I think I will not be expanding on this. If anyone wants to though, feel free to do it later. Regarding keeping the blacklist I think there are arguments for both ways. While a whitelist for what files to delete would be safer, it could also break the installation process if not configured correctly, and if you wanted to make a backup of some files, you are probably just better off making a backup of the server/container itself and keeping/deleting what you need.

The way it is setup it guarantees a reinstallation and keeps the world + important server configs and the JourneyMap which should get you set up in most cases.

samsonsin committed:
Modified update part of startup script + server start arguments

This changes how files are deleted / preserved during server update. This version specifically deletes files which need to be deleted according to docs, and preserves those that should be preserved (the preserve variable essentially excludes files from being deleted by the delete section, and if overwriting was enabled during unzip, they would still be preserved).

Also changed the -xms argument according to gtnh recommendations, which should likely be obeyed over minecraft standards since the pack uses a non-standard java for this minecraft version.

Added 2 variables which dictate this new functionality, so end users can easily change behaviour should it be needed in the future.

samsonsin committed:
Update java/gtnh/egg-greg-tech--new-horizons.json

Co-authored-by: DJSchaffner <Jonas.Wilkens@outlook.com>

samsonsin committed:
Update java/gtnh/egg-greg-tech--new-horizons.json

Co-authored-by: DJSchaffner <Jonas.Wilkens@outlook.com>

samsonsin committed:
Update java/gtnh/egg-greg-tech--new-horizons.json

Co-authored-by: DJSchaffner <Jonas.Wilkens@outlook.com>

samsonsin committed:
Update java/gtnh/egg-greg-tech--new-horizons.json

Co-authored-by: DJSchaffner <Jonas.Wilkens@outlook.com>

samsonsin committed:
Update java/gtnh/egg-greg-tech--new-horizons.json

Co-authored-by: DJSchaffner <Jonas.Wilkens@outlook.com>

DJSchaffner committed:
Merge pull request #2 from samsonsin/Possible-change-improvement-to-gtnh-egg

Modified update part of startup script + server start arguments

DJSchaffner committed:
Rename original egg to pterodactyl and add pelican egg
